src/pages/Category/Category.js

```diff
@@ -10,31 +9,54 @@ const Category = () => {
   const [modalSwitcher, setModalSwitcher] = useState(-1);
   const [modalStatus, setModalStatus] = useState(false);
 
-  const handleShowModal = () => {
-    setModalSwitcher(movieOption.id);
-    setModalStatus(!modalStatus);
-  };
+  const [pins, setPins] = useState([]);
+  const [page, setPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+
+  const fetchPins = async (page) => {
+    const res = await fetch(`http://10.58.52.52:3000/movies/category?page=${page}`);
+    const data = await res.json();
+    setPins((prev) => [...prev, ...data.movieCategory]);
 
-  const target = useRef(null);
+    setLoading(true);
+  };
 
   useEffect(() => {
-    const observer = new IntersectionObserver(callback, finScroll);
-    observer.observe(target.current);
-  }, [target]);
+    fetchPins(page);
+  }, [page]);
 
-  const finScroll = {
-    threshold: 1.0,
+  const loadMore = () => {
+    setPage((prev) => prev + 1);
   };
 
-  const callback = () => {
-    target = console.log('load');
+  const pageEnd = useRef();
+
+  useEffect(() => {
+    if (loading) {
+      const observer = new IntersectionObserver(
+        (entries) => {
+          console.log();
+          if (entries[0].isIntersecting) {
+            loadMore();
+          }
+        },
+        { threshold: 0.7 }
+      );
+      observer.observe(pageEnd.current);
+    }
```

Comment:
```suggestion
    if (!loading) return;
    const observer = new IntersectionObserver(
      (entries) => {
        if (entries[0].isIntersecting) {
          loadMore();
        }
      },
      { threshold: 0.7 }
    );
    observer.observe(pageEnd.current);
```

이런 식으로 하면 if문에 의한 nesting이 조금은 줄겠네요 👀
근데 `loading`이라는 이름과 다르게 `loading`이 `true`인 상태가 데이터가 다 불러와진 다음 아닌가요?